generics/unique.go

```diff
@@ -0,0 +1,30 @@
+//go:build go1.18
+
+package generics
+
+// Unique creates a map with keys being the values of a slice
+func Unique[T comparable](s []T) map[T]struct{} {
```

Comment:
should we call it `Set` instead? To follow the convention we created?

Reply:
My two cents, Should be `NewSet` then, you're creating a Set from a collection.

Though, `Unique` or `Uniq` (ruby <3) is fine.